dds/idl/be_global.cpp

```diff
@@ -748,12 +766,107 @@ ExtensibilityKind BE_GlobalData::extensibility(AST_Decl* node) const
   return value;
 }
 
+AutoidKind BE_GlobalData::autoid(AST_Decl* node) const
+{
+  AutoidAnnotation* autoid_annotation =
+    dynamic_cast<AutoidAnnotation*>(
+      builtin_annotations_["::@autoid"]);
+  AutoidKind value = autoid_annotation->absent_value;
+  autoid_annotation->node_value_exists(node, value);
+  return value;
+}
+
+bool BE_GlobalData::id(AST_Decl* node, ACE_CDR::ULong& value) const
+{
+  IdAnnotation* id_annotation =
+    dynamic_cast<IdAnnotation*>(builtin_annotations_["::@id"]);
+  return id_annotation->node_value_exists(node, value);
+}
+
+bool BE_GlobalData::hashid(AST_Decl* node, std::string& value) const
+{
+  HashidAnnotation* hashid_annotation =
+    dynamic_cast<HashidAnnotation*>(builtin_annotations_["::@hashid"]);
+  return hashid_annotation->node_value_exists(node, value);
+}
+
+bool BE_GlobalData::is_optional(AST_Decl* node) const
+{
+  OptionalAnnotation* optional_annotation =
+    dynamic_cast<OptionalAnnotation*>(
+      builtin_annotations_["::@optional"]);
+  bool value = optional_annotation->absent_value;
+  optional_annotation->node_value_exists(node, value);
+  return value;
+}
+
+bool BE_GlobalData::is_must_understand(AST_Decl* node) const
+{
+  MustUnderstandAnnotation* must_understand_annotation =
+    dynamic_cast<MustUnderstandAnnotation*>(
+      builtin_annotations_["::@must_understand"]);
+  bool value = must_understand_annotation->absent_value;
+  must_understand_annotation->node_value_exists(node, value);
+  return value;
+}
+
+bool BE_GlobalData::is_key(AST_Decl* node) const
+{
+  KeyAnnotation* key_annotation =
+    dynamic_cast<KeyAnnotation*>(
+      builtin_annotations_["::@key"]);
+  bool value = key_annotation->absent_value;
+  key_annotation->node_value_exists(node, value);
+  return value;
```

Comment:
```suggestion
  bool node_is_key;
  check_key(node, node_is_key);
  return value;
```

Reply:
I prefer the exist coding as it is structurally more similar to similar functions.

Reply:
The similar functions don't have a near duplicate existing function like this one does. If you don't want to use `check_key` directly, please define this function in terms of `check_key`. It doesn't make sense for there more than one code path for this.